xvec/accessor.py

```diff
@@ -918,6 +918,375 @@ def to_geodataframe(
         )
         return df
 
+    def agg_geom(
+        self,
+        geom,
+        trans,
+        var: str,
+        x_coords: str = None,
+        y_coords: str = None,
+        x_axis: int = None,
+        y_axis: int = None,
+        stat: str = "mean",
+        dask: bool = False,
+    ):
+        """Aggregate the values from a dataset over a polygon geometry.
+
+        The CRS of the raster and that of points need to be in wgs84.
+        Xvec does not verify their equality.
+
+        Parameters
+        ----------
+        geom : Polygon[shapely.Geometry]
+            An arrray-like (1-D) of shapely geometry, like a numpy array or GeoPandas
+            GeoSeries.
+        trans : affine.Affine
+            Affine transformer.
+            Representing the geometric transformation applied to the data.
+        x_coords : Hashable
+            Name of the axis containing ``x`` coordinates.
+        y_coords : Hashable
+            Name of the axis containing ``y`` coordinates.
+        x_axis : int
+            Axis order of  ``x`` coordinates.
+        y_axis : int
+            Axis order of  ``y`` coordinates.
+        var : Hashable
+            Name of the variable in the dataset to aggregate its values.
+        stat : Hashable
+            Spatial aggregation statistic method, by default "mean". It supports the
+            following statistcs: ['mean', 'median', 'min', 'max', 'sum']
+        dask : bool,
+            If the input is dask array or not.
+
+        Returns
+        -------
+        Array
+            Aggregated values over the geometry.
+
+        """
+        try:
+            import geopandas as gpd
+        except ImportError as err:
+            raise ImportError(
+                "The geopandas package is required for `xvec.agg_geom()`. "
+                "You can install it using 'conda install -c conda-forge geopandas' or "
+                "'pip install geopandas'."
+            ) from err
+
+        try:
+            import rasterio
+        except ImportError as err:
+            raise ImportError(
+                "The rasterio package is required for `xvec.agg_geom()`. "
+                "You can install it using 'conda install -c conda-forge rasterio' or "
+                "'pip install rasterio'."
+            ) from err
+
+        try:
+            import gc
+        except ImportError as err:
+            raise ImportError(
+                "The gc package is required for `xvec.agg_geom()`. "
+                "Make sure 'gc' is included in the standard library"
+                "Check your Python installation for any issues."
+            ) from err
+
+        # Create a GeoSeries from the geometry
+        geo_series = gpd.GeoSeries(geom)
+
+        # Convert the GeoSeries to a GeometryArray
+        geometry_array = geo_series.geometry.array
```

Comment:
```suggestion
        geometry_array = np.asarray(geom)
```

As mentioned above, we don't need to use geopandas.

Reply:
I've recheck it and found that this step not important and removed it completely.